bin/changelog

```diff
@@ -0,0 +1,303 @@
+    type: 'confirm',
+    message,
+    default: true
+  }) : true;
+
+  if (shouldProceed === false) {
+    process.exit(1);
+  }
+};
+
+/**
+ * Checks if the changelog entry has all the necessary properties. Returns a
+ * new object with only the changelog entry properties that can be safely
+ * stringified and put into a file. If the input object is missing properties
+ * or they're in invalid format, throws an error.
+ *
+ * @param {object} input A valid changelog entry object.
+ * @returns {object} The `input` object but only with valid changelog entry properties.
+ */
+const assertChangelogEntryFormat = (input) => {
+  if (typeof input !== 'object' || input === null) {
+    throw new TypeError(`input must be an object and not be null (got: ${input})`);
+  }
+
+  if (typeof input.title !== 'string' || input.title.length === 0) {
+    throw new TypeError(`input.title must be a non-empty string (got: ${input.title})`);
+  }
+
+  if (typeof input.type !== 'string' || !changelogEntryTypes.includes(input.type)) {
+    throw new TypeError(`input.type must be one of: ${changelogEntryTypes} (got: ${input.type})`);
+  }
+
+  if (typeof input.issue !== 'number' || Number.isNaN(input.issue)) {
+    throw new TypeError(`input.issue must be a non-NaN number (got: ${input.issue})`);
+  }
+
+  if (typeof input.framework !== 'string' || !changelogFrameworkTypes.includes(input.framework)) {
+    throw new TypeError(`input.framework must be one of: ${changelogFrameworkTypes} (got: ${input.framework})`);
+  }
+
+  return {
+    title: input.title,
+    type: input.type,
+    issue: input.issue,
+    framework: input.framework
+  };
+};
+
+const createEntryCommand = async(args) => {
+  // https://stackoverflow.com/a/11958481
+  const gitBranchName = (await execa('git', ['rev-parse', '--symbolic-full-name', '--abbrev-ref', 'HEAD'])).stdout;
+  const maybeExtractedNumber = extractNumberFromString(gitBranchName);
+
+  const answers = isTTY ? await inquirer.prompt([
+    {
+      name: 'title',
+      type: 'input',
+      message: 'Title of the entry (e.g. "Fixed an issue with...")',
+      validate: s => s.length > 0,
+      when: typeof args.title === 'undefined'
+    },
+    {
+      name: 'issue',
+      type: 'input',
+      message: '# of the related issue/PR (e.g. 512)',
+      default: maybeExtractedNumber,
+      validate: s => ((Number.isNaN(Number(s)) || s.length === 0) ? 'The provided input is not a number' : true),
+      filter: s => Number(s),
+      when: typeof args.issue === 'undefined'
+    },
+    {
+      name: 'type',
+      type: 'list',
+      message: 'Type of the change',
+      choices: changelogEntryTypes,
+
+      default: ({ title }) =>
+        // Attempts to find a suitable default choice based on the title.
+        // e.g. the title "Fixed every bug" would return the type "fixed",
+        // and "Added 10 features" would return "added".
+        changelogEntryTypes.find(
+          type => title.toLowerCase().includes(type)
```

Comment:
I get an error here:

<img width="1113" alt="Screenshot 2021-01-04 at 21 57 17" src="https://user-images.githubusercontent.com/566463/103578990-d1386000-4ed7-11eb-9df4-c19968e25ac6.png">

Reply:
Good catch, thanks https://github.com/handsontable/handsontable/pull/7448/commits/a7c90dd1ff1be9923bf1d1c708fc58a644f54448

Reply:
Hmm, after the changes I get a new error. The same applies to 
```sh
bin/changelog entry "Fixed an issue with sorting" --issue=444 --framework=none --type=fixed
```
and
```sh
bin/changelog entry Fixed an issue with sorting --issue=444 --framework=none --type=fixed
```

Error:
![image](https://user-images.githubusercontent.com/571316/103897596-1a64fb80-50f4-11eb-9dde-7a615259a9ac.png)

Reply:
I hate JavaScript sometimes, thanks https://github.com/handsontable/handsontable/pull/7448/commits/38febb738ca7ea25fce3896bda4b7a065157a696